File: rust/src/286.rs
//268. Missing Number
impl Solution {
    pub fn missing_number(nums: Vec<i32>) -> i32 {
        let n = nums.len() as i32;
        let expected_sum = (n * (n + 1)) / 2;
        let actual_sum: i32 = nums.iter().sum();

        expected_sum - actual_sum
    }
}
//sub: https://leetcode.com/problems/missing-number/submissions/1570881819/
//sol: https://leetcode.com/problems/missing-number/solutions/6526404/268-missing-number-by-nalindalal2004-hpbp/

File: ts/src/286.ts
//2294. Partition Array Such That Maximum Difference Is K
function partitionArray(nums: number[], k: number): number {
  nums.sort((a, b) => a - b); // Sort ascending

  let count = 1;
  let start = nums[0];

  for (let i = 1; i < nums.length; i++) {
    if (nums[i] - start > k) {
      count++;
      start = nums[i];
    }
  }

  return count;
}
//sub: https://leetcode.com/problems/partition-array-such-that-maximum-difference-is-k/submissions/1669025400/?envType=daily-question&envId=2025-06-19
//sol: https://leetcode.com/problems/partition-array-such-that-maximum-difference-is-k/solutions/6860292/2294-partition-array-such-that-maximum-d-16ky/
